llvm/lib/Target/AArch64/GISel/AArch64InstructionSelector.cpp

```diff
@@ -1994,7 +1994,108 @@ bool AArch64InstructionSelector::selectVectorAshrLshr(
 
 bool AArch64InstructionSelector::selectVaStartAAPCS(
     MachineInstr &I, MachineFunction &MF, MachineRegisterInfo &MRI) const {
-  return false;
+
+  if (MF.getSubtarget<AArch64Subtarget>().isCallingConvWin64(
+          MF.getFunction().getCallingConv(), MF.getFunction().isVarArg()))
```

Comment:
Is test over specified? `isVarArg()` is implied by the function name. What about `isWindowsArm64EC()` ?  What about `isTargetWindows()` ? Or is it under specified?

Reply:
Sorry. Could you instead copy the SDAG and do the Windows test here:
https://github.com/llvm/llvm-project/blob/350019e8b178394de03ff0cab38375ba806c9dc0/llvm/lib/Target/AArch64/GISel/AArch64InstructionSelector.cpp#L3564
and add an empty `selectWin64VASTART` function, which returns false.

Reply:
In C `__attribute__((ms_abi))` can be used to enable Win64 ABI. `selectVaStartDarwin` handles this property. See commit https://github.com/llvm/llvm-project/commit/1c10d5b175992a9d056a2d763a932e5652386fc1.

This `if` was copied from darwin. I don't know the exact details on what this attribute implies as well as Win64 ABI, so I'm leaving this unsupported.

I think we shouldn't touch `case TargetOpcode::G_VASTART: ` to specifically handle Win64 in this PR.